Core/HLE/sceNet.cpp

```diff
@@ -124,19 +124,8 @@ static int InitLocalhostIP() {
 	// If lookup fails, we'll assume it's not local.
 	isLocalServer = false;
 
-	addrinfo *resultAddr = nullptr;
-	std::string error;
-	if (net::DNSResolve(g_Config.proAdhocServer, "", &resultAddr, error, net::DNSType::IPV4)) {
-		for (addrinfo *ptr = resultAddr; ptr != nullptr; ptr = ptr->ai_next) {
-			auto addr4 = ((sockaddr_in *)ptr->ai_addr)->sin_addr.s_addr;
-			isLocalServer = (ntohl(addr4) & 0x7F000000) == 0x7F000000;
-		}
-
-		net::DNSResolveFree(resultAddr);
-		resultAddr = nullptr;
-	}
-
-	return 0;
+	std::string serverStr = StringLower(StripSpaces(g_Config.proAdhocServer));
+	isLocalServer = (serverStr == "localhost" || serverStr.find("127.") == 0);
```

Comment:
It's generally always more efficient to do an insensitive compare rather than change something to do an insensitive compare.  I'd suggest just `!strcasecmp(g_Config.proAdhocServer.c_str(), "localhost")`.

IMHO, we shouldn't have non-Unicode upper/lower funcs because it gives you a false sense of security (I realize we have some but they're unused.)  With a strcasecmp against a constant, it's much more obvious that you're correctly handling Unicode.

Should also remove the ` = false;` above because it's a dead store now.

-[Unknown]